astropy/convolution/tests/test_convolve_fft.py

```diff
@@ -606,13 +653,15 @@ def test_uniform_3x3_withnan(self, boundary, nan_treatment,
                               "convolve boundary='fill'"):
                 z = convolve_fft(x, y, boundary=boundary,
                                  nan_treatment=nan_treatment,
-                                 fill_value=np.nan if normalize_kernel else 0,
+                                 # you cannot fill w/nan, you can only interpolate over it
+                                 fill_value=np.nan if normalize_kernel and nan_treatment=='interpolate' else 0,
```

Comment:
Even though this was already conditional on ``normalize_kernel``, I don't think it should be constraining the test space. The addition of ``nan_treatment=='interpolate'`` also constrains the test and I don't think it is a good idea to do so. We should be trying to test all parameter space and just deal with the outcome appropriately in the test.

Reply:
I've lost the thread of this conversation... are you saying we should run this test with both `np.nan` and `0` for fill values, and just check the returns for each case are correct?  That's reasonable, and I think the expected outcome is:
`if np.isnan(fill_value) and not (normalize_kernel and nan_treatment=='interpolate'): an exception should be raised`, but all cases should be allowed with `fill_value=0`.

I'm not certain of this, though, I should go check....